packages/circuit-ui/components/NotificationModal/NotificationModal.tsx

```diff
@@ -0,0 +1,228 @@
+            outline: none;
+            background-color: ${theme.colors.white};
+            border-radius: ${theme.borderRadius.mega};
+            padding: ${theme.spacings.giga};
+            text-align: center;
+            opacity: 0;
+            transition: opacity ${TRANSITION_DURATION}ms ease-in-out;
+            overflow-y: auto;
+
+            ${theme.mq.untilKilo} {
+              -webkit-overflow-scrolling: touch;
+            }
+          `,
+
+          afterOpen: cssString`
+           opacity: 1;
+          `,
+          beforeClose: cssString`
+            opacity: 0;
+        `,
+        };
+
+        const overlayStyles = {
+          base: cssString`
+            position: fixed;
+            top: 0;
+            left: 0;
+            bottom: 0;
+            right: 0;
+            opacity: 0;
+            transition: opacity ${TRANSITION_DURATION}ms ease-in-out;
+            background: ${theme.colors.overlay};
+            z-index: ${theme.zIndex.modal};
+
+            ${theme.mq.kilo} {
+              -webkit-overflow-scrolling: touch;
+              overflow-y: auto;
+            }
+          `,
+          afterOpen: cssString`
+            opacity: 1;
+          `,
+          beforeClose: cssString`
+            opacity: 0;
+          `,
+        };
+
+        const reactModalProps = {
+          className: styles,
+          overlayClassName: overlayStyles,
+          onRequestClose: handleClose,
+          closeTimeoutMS: TRANSITION_DURATION,
+          shouldCloseOnOverlayClick: !preventClose,
+          shouldCloseOnEsc: !preventClose,
+          ...props,
+        };
+
+        function wrapOnClick(onClick?: ButtonProps['onClick']) {
+          return (event: ClickEvent) => {
+            handleClose?.(event);
+            onClick?.(event);
+          };
+        }
+
+        return (
+          <ReactModal {...reactModalProps}>
+            {!preventClose && closeButtonLabel && (
+              <CloseButton
+                onClick={onClose}
+                label={closeButtonLabel}
+                css={closeButtonStyles}
+              />
+            )}
+            <ModalImage {...image} />
+            <Headline
+              as="h2"
+              size="three"
+              css={spacing({ top: 'giga', bottom: 'byte' })}
+              noMargin
+            >
+              {headline}
+            </Headline>
+            <Body noMargin>{body}</Body>
+            {actions && (
+              <ButtonGroup
+                actions={{
+                  primary: actions?.primary && {
+                    ...actions.primary,
+                    onClick: wrapOnClick(actions.primary.onClick),
+                  },
+                  secondary: actions?.secondary && {
+                    ...actions.secondary,
+                    onClick: wrapOnClick(actions.secondary.onClick),
+                  },
+                }}
+                css={spacing({ top: 'giga' })}
```

Comment:
This is a massive nitpick but it would seem more "natural" to me if we added the spacing in the Body since that's where we're also setting `noMargin`. But it won't matter anymore #AfterWeRemoveTheNoMargin

Reply:
The body is optional though, so we can't rely on it and its spacing being present.

Reply:
ahaaa true, well okay then 🤷

Reply:
Which reminds me, I just realised that both body and image were required in the implementation. I made them optional, will push changes now.